xbundle/__init__.py

```diff
@@ -271,52 +271,54 @@ def import_xml_removing_descriptor(self, path, xml):
         url_name = xml.get('url_name', '')
         if xml.tag in DESCRIPTOR_TAGS and 'url_name' in xml.attrib and url_name:
             # XML stores path separator as a colon.
+            print("xml.tag: " + xml.tag + " is in DESCRIPTOR_TAGS and url_name (" + url_name + ") is in xml.attrib;")
             unfn = url_name.replace(':', '/')
             filename = join(path, xml.tag, (unfn + '.xml'))
-            if not exists(filename):
-                return xml
```

Comment:
This appears to have been short circuiting the recursive function -- abandoning all children of xml nodes which had url_name attributes but no corresponding file on disk.